standalone-metastore/metastore-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogServer.java

```diff
@@ -0,0 +1,169 @@
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.IOException;
+import java.lang.ref.Reference;
+import java.lang.ref.SoftReference;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.servlet.http.HttpServlet;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.SecureServletCaller;
+import org.apache.hadoop.hive.metastore.ServletSecurity;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.iceberg.HiveCachingCatalog;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.eclipse.jetty.server.ConnectionFactory;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.gzip.GzipHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class HMSCatalogServer {
+  private static final String CACHE_EXPIRY = "hive.metastore.catalog.cache.expiry";
+  private static final String JETTY_THREADPOOL_MIN = "hive.metastore.catalog.jetty.threadpool.min";
+  private static final String JETTY_THREADPOOL_MAX = "hive.metastore.catalog.jetty.threadpool.max";
+  private static final String JETTY_THREADPOOL_IDLE = "hive.metastore.catalog.jetty.threadpool.idle";
+  private static final Logger LOG = LoggerFactory.getLogger(HMSCatalogServer.class);
+  private static Reference<Catalog> catalogRef;
+
+  static Catalog getLastCatalog() {
+    return catalogRef != null ? catalogRef.get() :  null;
+  }
+
+  private HMSCatalogServer() {
+    // nothing
+  }
+
+  public static HttpServlet createServlet(SecureServletCaller security, Catalog catalog) throws IOException {
+    return new HMSCatalogServlet(security, new HMSCatalogAdapter(catalog));
+  }
+
+  public static Catalog createCatalog(Configuration configuration) {
+    final String curi = configuration.get(MetastoreConf.ConfVars.THRIFT_URIS.getVarname());
+    final String cwarehouse = configuration.get(MetastoreConf.ConfVars.WAREHOUSE.getVarname());
+    final String cextwarehouse = configuration.get(MetastoreConf.ConfVars.WAREHOUSE_EXTERNAL.getVarname());
+    MetastoreConf.setVar(configuration, MetastoreConf.ConfVars.THRIFT_URIS, "");
+    MetastoreConf.setVar(configuration, MetastoreConf.ConfVars.HMS_HANDLER_CREATE, "newHMSRetryingLocalHandler");
+    final HiveCatalog catalog = new org.apache.iceberg.hive.HiveCatalog();
+    catalog.setConf(configuration);
+    Map<String, String> properties = new TreeMap<>();
+    if (curi != null) {
+      properties.put("uri", curi);
```

Comment:
the curi should always be empty?

Reply:
In the current 'HMS embedded' case, yes; let me add a clarifying comment.